binding/form_mapping.go

```diff
@@ -198,6 +199,14 @@ func setByForm(value reflect.Value, field reflect.StructField, form map[string][
 }
 
 func setWithProperType(val string, value reflect.Value, field reflect.StructField) error {
+	if value.Kind() != reflect.Struct {
```

Comment:
Why would you need this condition? I think it's perfectly fine to have structs that implement `TextUnmarshaler`.

Reply:
Also, I would consider checking both `value` and `value.Addr`. It's possible that someone is already using a pointer in the struct that you're binding into:
```go
var out struct {
    MyParam *MyType `form:"param"`
}
```

Reply:
I think this is leftover from when I was trying to reuse the TextUnmarshaler interface, but that was also implemented by some go structs (`time.Time`?), which was causing problems. I think that's why we switched to UnmarshalParam.. so maybe this isn't needed anymore

Reply:
I added a test for the pointer type and it seems to work as is